### ISSUE TITLE
Asset registration doesn't work with a remote Jupyter server

### ISSUE THREAD

**Issue by joshuamanns** [JuliaGizmos/AssetRegistry.jl#10]
When using a library like `Interact` that relies on `AssetRegistry` on a remote Jupyter server, 404s are thrown or assets.
See this thread: https://github.com/JuliaGizmos/Interact.jl/issues/270

**Comment by shashi** [comment]
Thanks for the issue! @travigd you needed to solve this problem too right?

One idea which can potentially work: I believe there's only a ZMQ link between the kernel and the Jupyter server. So we can make our plugin serve files by fetching it via ZMQ instead of directly reading from disk. IMO this is still much better / proper than trying to send files over WebSocket by writing some Julia code.

**Comment by travigd** [comment]
@shashi This is a little bit different than what I was talking about (which is running a Jupyter **kernel** on a remote server rather than the entire notebook/lab server).

At any rate, are you using JupyterHub? What is the url of your notebook and what URL is the AssetRegistry using (_i.e._ what URL do you see 404 errors for in the browser console)?

**Comment by shashi** [comment]
Oh I assumed "server" means kernel in the issue title...

**Comment by travigd** [comment]
Yeah, remote kernel (insofar as remote to the web server) is a pretty rare case as far as I know. The only common case is "enterprise" kernels (i.e. things that run on clusters like Spark).

@joshuamanns Are you still having issues?

**Comment by ilanpillemer** [comment]
I am having this issue

**Comment by travigd** [comment]
Is it the double-slash that's causing the issue? I've not had these kinds of issues on JupyterHub myself.

**Comment by ilanpillemer** [comment]
when I tried removing the double slash and pasting into the url bar, it still didnt give me the js file.

`ilan-pillemer-brambles-dev@jupyter-ilan-pillemer-brambles-dev:/home/mount$ echo $JUPYTERHUB_SERVICE_PREFIX
/jupyter-dev/user/ilan.pillemer@brambles@dev/
ilan-pillemer-brambles-dev@jupyter-ilan-pillemer-brambles-dev:/home/mount$ echo $JULIA_ASSETREGISTRY_BASEURL
ilan-pillemer-brambles-dev@jupyter-ilan-pillemer-brambles-dev:/home/mount$ echo $ENV
ilan-pillemer-brambles-dev@jupyter-ilan-pillemer-brambles-dev:/home/mount$`

So the double slash is caused because `JUPYTERHUB_SERVICE_PREFIX` has a trailing slash.
According to this https://github.com/jupyterhub/jupyterhub/issues/997 its always meant to have a trailing slash if it exists.

But when I simply tried pasting the URL without the double slash it still seemed to not work.
So I think its definitely part of the problem, but maybe not all of the problem.

Do you know how I can diagnose to see what the URL it is expecting should be? ie where the asset server is running on?

**Comment by travigd** [comment]
Is WebIO pre-installed on the server?

The assetregistry extension is a Jupyter server extension, which needs to be installed *before* the Jupyter notebook instance starts (there's no way to activate it after the Jupyter notebook has started).

**Comment by ilanpillemer** [comment]
Currently I run 

```
using Plots
using Interact
using WebIO
WebIO.install_jupyter_nbextension
```

or alternatively the lab extension.

and then restart the kernel.

currently I believe we use this to start-up the Julia kernel in the k8s cluster as part of Jupyter Hub running the K8s cluster.

```
RUN julia -e 'import Pkg; Pkg.update()' && \
    (test $TEST_ONLY_BUILD || julia -e 'import Pkg; Pkg.add("HDF5")') && \
    julia -e "using Pkg; pkg\"add IJulia\"; pkg\"precompile\""
```

I would assume we would need to change this to include `WebIO` and `Interact`. Any suggestion on the best way to do that?

**Comment by travigd** [comment]
Yep, just add WebIO and run `WebIO.install_jupyter_nbextension()` as part of that same run.

```dockerfile
RUN julia -e 'import Pkg; Pkg.update()' && \
    (test $TEST_ONLY_BUILD || julia -e 'import Pkg; Pkg.add("HDF5")') && \
    julia -e '\
        using Pkg; pkg"add IJulia WebIO"; pkg"precompile"; \
        using WebIO; WebIO.install_jupyter_nbextension(); \
    '
```

**Comment by ilanpillemer** [comment]
that made all the difference.

![image](https://user-images.githubusercontent.com/892046/96180315-28a6df80-0f2a-11eb-8046-325c702a3bde.png)

However I still needed to run the below in the notebook and then type `CMD-R`.. but after that it worked 🎉 

Anyway to avoid having to do that?

```
using WebIO
WebIO.install_jupyter_nbextension
```
